product_packaging_dimension/migrations/14.0.1.0.0/pre-migrate.py

```diff
@@ -5,18 +5,9 @@
 
 
 def migrate(cr, version):
-    # Rename lnght into packaging_length
-    if column_exists(cr, "product_packaging", "lnght"):
-        rename_column(cr, "product_packaging", "lnght", "packaging_length")
```

Comment:
There was an issue there, the column is named `lngth`, not `lnght` that's why we didn't get the values during the migration. We need to move this script to `migrations/14.0.2.0.0/pre-migrate.py` with the right column name (no need to keep the current one).
As by default the length UoM used is meter in v14, the script is probably right to divide the values by 1000 as it was in millimeters in v13. But our custom need is to keep millimeters in v14, should we use a environment variable to change the behavior of this script to not divide the values by 1000? This same environment variable will be used elsewhere where we were storing millimeters values (like in `stock_storage_type`, the migration needs to be checked there as well).
@i-vyshnevska @simahawk What do you think?

Reply:
@sebalix  I remember the story about "length" "lngth" and in core, it was renamed to "packagin_length" so there is no need to have "lngth" anymore

I changed migration in update 14.0.1.0.0 because it was done without considering of change int -> float and all data is lost so for projects migrated from now it should land for that step (data was lost for width, height as well)

for providing script to consider a change of accuracy with considering of uom - not all of uom are coming from core, for instance, mm defined in "stock_measuring_device" some can be set manually so add them can be tricky

Reply:
I wasn't talking about the `length` and `lngth` story, but the existing typo in the migration script where the last two letters were inverted: `lngth` vs `lnght`, so the script could never work as it couldn't pass the `if column_exists()` condition.

Now I still do not understand why we should delete the conversion of millimeters to meters in this OCA module.

- in v13 the `width` field was hosting millimeters value (no related UoM to change that)
- in v14 the business rule from Odoo is to enforce this kind of fields to meters thanks to the `product.template._get_length_uom_id_from_ir_config_parameter()`), and all OCA v14 addons are now based on this rule (see the different `length_uom_id` field in `stock.location.storage.type`, `product.packaging`, `stock.quant.package`...), so by default why not converting the existing millimeter values to meter? I know we need millimeters in our project but it's up to us to change that, in OCA we should stick to sane & standard defaults.

Reply:
- data was lost not only for lngth for other dimensions as well, so not only typo is the reason of data lost
- my idea is on v13 it was explicitly called mm for all 3 dimensions so why that should change
- as the field should stay integer we really will have losing of accuracy on width/1000 as if you have 200 mm  on converting to m in integer we will have 0 
- we don't have an mm uom in here and suppose we don't want to bring a dependency here

my point is to keep data untouched and the rest will be handled client by client in dependency on their configuration, which will just require a change of length_uom_id

Reply:
Ok it's clearer, thanks!
That means that currently in core Odoo we could only expressed dimensions of a packaging in meters or foots, with entire values? That's crazy, I miss something...

I also wanted to create a `uom_mm` addon few months ago for this kind of dependency (rejected here https://github.com/OCA/product-attribute/pull/914), together with [product_logistics_uom](https://github.com/OCA/product-attribute/tree/14.0/product_logistics_uom) it could have been possible to define the millimeter as the standard UoM for logistics. Having this UoM in `stock_measuring_device` is not its place, we now have to depend on this addon everywhere we need to reference `mm`... Anyway.

Reply:
> wanted to create a uom_mm addon few months ago for this kind of dependency (rejected here #914), 

We should probably reopen this discussion :wink: